src/ocamlbuild_where.ml

```diff
@@ -11,10 +11,27 @@
 (*                                                                     *)
 (***********************************************************************)
 
-let bindir = ref Ocamlbuild_config.bindir;;
+module O = Ocamlbuild_config;;
+
+let bindir = ref O.bindir;;
 let libdir = ref begin
-  Filename.concat
-    (try Sys.getenv "OCAMLLIB"
-     with Not_found -> Ocamlbuild_config.libdir)
-    "ocamlbuild"
+  let root, suffix =
+    let ocaml_lib_len = String.length O.ocaml_libdir + 1 in
+    let lib_len = String.length O.libdir_abs in
+    if lib_len < ocaml_lib_len then
+      O.libdir, "ocamlbuild"
+    else
+      let prefix =
+        String.sub O.libdir_abs 0 ocaml_lib_len
+      in
+      if prefix = O.ocaml_libdir ^ "/" then
```

Comment:
Wouldn't it be more robust to use `O.ocaml_libdir ^ Filename.dir_sep`, and compute the length of this for the `String.sub` call?